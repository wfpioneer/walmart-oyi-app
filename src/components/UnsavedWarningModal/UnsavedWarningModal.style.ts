import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

const styles = StyleSheet.create({
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    alignContent: 'center',
    height: 80,
    elevation: 10
  },
  buttonAlign: {
    alignSelf: 'center',
    width: '40%',
    margin: 10
  },
  message: {
    textAlign: 'center',
    fontSize: 16,
    padding: 15
  },
  labelHeader: {
    fontSize: 18,
    textAlign: 'center',
    color: COLOR.RED_500,
    paddingBottom: 5
  }
});

export default styles;
