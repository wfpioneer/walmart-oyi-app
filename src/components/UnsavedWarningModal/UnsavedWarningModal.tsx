import React from 'react';
import { Text, View } from 'react-native';
import { Dispatch } from 'redux';
import { NavigationProp } from '@react-navigation/native';
import { CustomModalComponent } from '../../screens/Modal/Modal';
import { strings } from '../../locales';
import { UseStateType } from '../../models/Generics.d';
import Button, { ButtonType } from '../buttons/Button';
import COLOR from '../../themes/Color';
import styles from './UnsavedWarningModal.style';

export const renderUnsavedWarningModal = (
  displayWarningModal: boolean,
  setDisplayWarningModal: UseStateType<boolean>[1],
  title: string,
  message: string,
  // eslint-disable-next-line @typescript-eslint/ban-types
  onConfirm: () => void
) => (
  <CustomModalComponent
    isVisible={displayWarningModal}
    onClose={() => setDisplayWarningModal(false)}
    modalType="Popup"
  >
    <>
      <View>
        <Text style={styles.labelHeader}>{title}</Text>
        <Text style={styles.message}>{message}</Text>
      </View>
      <View style={styles.buttonContainer}>
        <Button
          style={styles.buttonAlign}
          title={strings('GENERICS.CANCEL')}
          titleColor={COLOR.MAIN_THEME_COLOR}
          type={ButtonType.SOLID_WHITE}
          onPress={() => setDisplayWarningModal(false)}
          testID="cancelBack"
        />
        <Button
          style={styles.buttonAlign}
          title={strings('GENERICS.OK')}
          type={ButtonType.PRIMARY}
          onPress={() => onConfirm()}
          testID="confirmBack"
        />
      </View>
    </>
  </CustomModalComponent>
);
