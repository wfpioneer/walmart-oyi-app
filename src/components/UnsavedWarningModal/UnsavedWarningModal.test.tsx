import { fireEvent, render } from '@testing-library/react-native';
import { renderUnsavedWarningModal } from './UnsavedWarningModal';

const mockSetState = jest.fn();
const mockOnConfirm = jest.fn();

describe('Unsaved warning modal render tests', () => {
  it('renders the warning modal', () => {
    const { toJSON, getByTestId } = render(renderUnsavedWarningModal(
      true,
      mockSetState,
      'yes',
      'no',
      mockOnConfirm
    ));

    const cancelBtn = getByTestId('cancelBack');
    const confirmBtn = getByTestId('confirmBack');

    fireEvent.press(cancelBtn);
    expect(mockSetState).toHaveBeenCalled();
    expect(mockOnConfirm).not.toHaveBeenCalled();

    jest.resetAllMocks();

    fireEvent.press(confirmBtn);
    expect(mockSetState).not.toHaveBeenCalled();
    expect(mockOnConfirm).toHaveBeenCalled();

    expect(toJSON()).toMatchSnapshot();
  });

  it('does not render the warning modal', () => {
    const { toJSON } = render(renderUnsavedWarningModal(
      false,
      mockSetState,
      'yes',
      'nein',
      mockOnConfirm
    ));

    expect(toJSON()).toMatchSnapshot();
  });
});
