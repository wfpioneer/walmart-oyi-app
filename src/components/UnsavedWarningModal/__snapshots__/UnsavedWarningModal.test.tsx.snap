// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unsaved warning modal render tests does not render the warning modal 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
/>
`;

exports[`Unsaved warning modal render tests renders the warning modal 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#00000099",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#ffffffff",
            "borderRadius": 7.5,
            "justifyContent": "space-between",
            "paddingVertical": 15,
            "width": "75%",
          },
          {
            "minHeight": undefined,
          },
        ]
      }
    >
      <View>
        <Text
          style={
            {
              "color": "#f44336",
              "fontSize": 18,
              "paddingBottom": 5,
              "textAlign": "center",
            }
          }
        >
          yes
        </Text>
        <Text
          style={
            {
              "fontSize": 16,
              "padding": 15,
              "textAlign": "center",
            }
          }
        >
          no
        </Text>
      </View>
      <View
        style={
          {
            "alignContent": "center",
            "elevation": 10,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "space-evenly",
          }
        }
      >
        <View
          style={
            [
              {
                "height": 45,
              },
              {
                "alignSelf": "center",
                "margin": 10,
                "width": "40%",
              },
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#ffffffff",
                  "borderColor": "#0071e9",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "flex": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
              testID="cancelBack"
            >
              <Text
                style={
                  {
                    "color": "#0071e9",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "textAlign": undefined,
                  }
                }
              >
                [missing "en.GENERICS.CANCEL" translation]
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "height": 45,
              },
              {
                "alignSelf": "center",
                "margin": 10,
                "width": "40%",
              },
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#0071e9",
                  "borderRadius": 5,
                  "elevation": 1,
                  "flex": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
              testID="confirmBack"
            >
              <Text
                style={
                  {
                    "color": "#ffffffff",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "textAlign": undefined,
                  }
                }
              >
                [missing "en.GENERICS.OK" translation]
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
